Typescript eslint complains about "object" in tools.ts

I generated a new adapter project using the latest create-adapter script (last week). When running GitHub actions for Test and Release, it fails.

According to eslint, one should no longer use `object` to describe an object, but rather `Record<string, unknown>`

Error message:
```text
/home/runner/work/ioBroker.loxone/ioBroker.loxone/src/lib/tools.ts
  7:26  warning  Argument 'it' should be typed with a non-any type                                                                                                                                                                                                              @typescript-eslint/explicit-module-boundary-types
  7:42  error    Don't use `object` as a type. The `object` type is currently hard to use ([see this issue](https://github.com/microsoft/TypeScript/issues/21732)).
Consider using `Record<string, unknown>` instead, as it allows you to more easily inspect and use the keys  @typescript-eslint/ban-types
 19:25  warning  Argument 'it' should be typed with a non-any type                                                                                                                                                                                                              @typescript-eslint/explicit-module-boundary-types

✖ 3 problems (1 error, 2 warnings)
```

Fixed `isObject()` function:
```typescript
export function isObject(it: any): it is Record<string, unknown> {
    // This is necessary because:
    // typeof null === 'object'
    // typeof [] === 'object'
    // [] instanceof Object === true
    return Object.prototype.toString.call(it) === '[object Object]';
}
```

--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -39,6 +39,13 @@ module.exports = {
 		"@typescript-eslint/no-object-literal-type-assertion": "off",
 		"@typescript-eslint/interface-name-prefix": "off",
 		"@typescript-eslint/no-non-null-assertion": "off", // This is necessary for Map.has()/get()!
+		"@typescript-eslint/no-inferrable-types": [
+			"error",
+			{
+				ignoreParameters: true,
+				ignoreProperties: true,
+			},
+		],
 	},
 	overrides: [
 		{

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,10 @@
 	(at the beginning of a new line )
 -->
 
+## __WORK IN PROGRESS__
+* (AlCalzone) Fixed ESLint errors in the generated `src/lib/tools.ts` (fixes #545)
+* (AlCalzone) Fixed ESLint errors in the creator sources
+
 ## 1.26.0 (2020-06-22)
 * (AlCalzone) The auto-deploy now also creates Github Releases
 * (AlCalzone) Modernized the main file templates with some best practices and async methods

--- a/package.json
+++ b/package.json
@@ -69,6 +69,7 @@
   "dependencies": {
     "@typescript-eslint/parser": "^2.27.0",
     "alcalzone-shared": "^2.2.0",
+    "ansi-colors": "^4.1.1",
     "axios": "^0.19.2",
     "debug": "^4.1.1",
     "enquirer": "^2.3.5",

--- a/src/lib/tools.ts
+++ b/src/lib/tools.ts
@@ -80,7 +80,6 @@ export function executeCommand(
 				options.stdout || process.stdout,
 				options.stderr || process.stderr,
 			],
-			// @ts-ignore This option exists starting with NodeJS 8
 			windowsHide: true,
 		};
 		if (options.cwd != null) spawnOptions.cwd = options.cwd;
@@ -348,6 +347,7 @@ export function tsFixQuotes(
 	const linter = new Linter();
 	linter.defineParser(
 		"@typescript-eslint/parser",
+		// eslint-disable-next-line @typescript-eslint/no-var-requires
 		require("@typescript-eslint/parser"),
 	);
 	const result = linter.verifyAndFix(

--- a/templates/_github/ISSUE_TEMPLATE/bug_report.md.ts
+++ b/templates/_github/ISSUE_TEMPLATE/bug_report.md.ts
@@ -1,6 +1,6 @@
 import { readFile, TemplateFunction } from "../../../src/lib/createAdapter";
 
-const templateFunction: TemplateFunction = answers => {
+const templateFunction: TemplateFunction = _answers => {
 	return readFile("bug_report.raw.md", __dirname);
 };
 

--- a/templates/admin/index_m.html.ts
+++ b/templates/admin/index_m.html.ts
@@ -1,7 +1,7 @@
 import { TemplateFunction } from "../../src/lib/createAdapter";
 import { AdapterSettings, getDefaultAnswer } from "../../src/lib/questions";
 
-function generateSettingsDiv(settings: AdapterSettings) {
+function generateSettingsDiv(settings: AdapterSettings): string {
 	if (settings.inputType === "select" && settings.options) {
 		const options = settings.options.map(opt => `
 					<option value="${opt.value}">${opt.text}</option>\n`);

--- a/templates/src/lib/tools.raw.ts
+++ b/templates/src/lib/tools.raw.ts
@@ -4,7 +4,7 @@ import axios from "axios";
  * Tests whether the given variable is a real object and not an Array
  * @param it The variable to test
  */
-export function isObject(it: any): it is object {
+export function isObject(it: unknown): it is Record<string, any> {
 	// This is necessary because:
 	// typeof null === 'object'
 	// typeof [] === 'object'
@@ -16,7 +16,7 @@ export function isObject(it: any): it is object {
  * Tests whether the given variable is really an Array
  * @param it The variable to test
  */
-export function isArray(it: any): it is any[] {
+export function isArray(it: unknown): it is any[] {
 	if (Array.isArray != null) return Array.isArray(it);
 	return Object.prototype.toString.call(it) === "[object Array]";
 }

--- a/templates/test/package.js.ts
+++ b/templates/test/package.js.ts
@@ -1,5 +1,5 @@
 import { readFile, TemplateFunction } from "../../src/lib/createAdapter";
 
-export = (answers => {
+export = (_answers => {
 	return readFile("package.raw.js", __dirname);
 }) as TemplateFunction;

--- a/test/baselines/TS_SingleQuotes/src/lib/tools.ts
+++ b/test/baselines/TS_SingleQuotes/src/lib/tools.ts
@@ -4,7 +4,7 @@ import axios from 'axios';
  * Tests whether the given variable is a real object and not an Array
  * @param it The variable to test
  */
-export function isObject(it: any): it is object {
+export function isObject(it: unknown): it is Record<string, any> {
 	// This is necessary because:
 	// typeof null === 'object'
 	// typeof [] === 'object'
@@ -16,7 +16,7 @@ export function isObject(it: any): it is object {
  * Tests whether the given variable is really an Array
  * @param it The variable to test
  */
-export function isArray(it: any): it is any[] {
+export function isArray(it: unknown): it is any[] {
 	if (Array.isArray != null) return Array.isArray(it);
 	return Object.prototype.toString.call(it) === '[object Array]';
 }

--- a/test/baselines/adapter_TS_ES6Class_ESLint_Tabs_DoubleQuotes_MIT/src/lib/tools.ts
+++ b/test/baselines/adapter_TS_ES6Class_ESLint_Tabs_DoubleQuotes_MIT/src/lib/tools.ts
@@ -4,7 +4,7 @@ import axios from "axios";
  * Tests whether the given variable is a real object and not an Array
  * @param it The variable to test
  */
-export function isObject(it: any): it is object {
+export function isObject(it: unknown): it is Record<string, any> {
 	// This is necessary because:
 	// typeof null === 'object'
 	// typeof [] === 'object'
@@ -16,7 +16,7 @@ export function isObject(it: any): it is object {
  * Tests whether the given variable is really an Array
  * @param it The variable to test
  */
-export function isArray(it: any): it is any[] {
+export function isArray(it: unknown): it is any[] {
 	if (Array.isArray != null) return Array.isArray(it);
 	return Object.prototype.toString.call(it) === "[object Array]";
 }

--- a/test/baselines/adapter_TS_ESLint_Tabs_DoubleQuotes_MIT/src/lib/tools.ts
+++ b/test/baselines/adapter_TS_ESLint_Tabs_DoubleQuotes_MIT/src/lib/tools.ts
@@ -4,7 +4,7 @@ import axios from "axios";
  * Tests whether the given variable is a real object and not an Array
  * @param it The variable to test
  */
-export function isObject(it: any): it is object {
+export function isObject(it: unknown): it is Record<string, any> {
 	// This is necessary because:
 	// typeof null === 'object'
 	// typeof [] === 'object'
@@ -16,7 +16,7 @@ export function isObject(it: any): it is object {
  * Tests whether the given variable is really an Array
  * @param it The variable to test
  */
-export function isArray(it: any): it is any[] {
+export function isArray(it: unknown): it is any[] {
 	if (Array.isArray != null) return Array.isArray(it);
 	return Object.prototype.toString.call(it) === "[object Array]";
 }

